service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/C21OrderController.java

```diff
@@ -0,0 +1,113 @@
+
+@Api
+@RequestMapping("/callback/create-order")
+@RestController
+public class C21OrderController {
+
+    private final ObjectMapper mapper;
+    private final CreateC21OrderService service;
+    private final ValidateGroupService validateGroupService;
+    private final DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+    private final UploadDocumentService uploadDocumentService;
+
+    @Autowired
+    public C21OrderController(ObjectMapper mapper,
+                              CreateC21OrderService service,
+                              ValidateGroupService validateGroupService,
+                              DocmosisDocumentGeneratorService docmosisDocumentGeneratorService,
+                              UploadDocumentService uploadDocumentService) {
+        this.mapper = mapper;
+        this.service = service;
+        this.validateGroupService = validateGroupService;
+        this.docmosisDocumentGeneratorService = docmosisDocumentGeneratorService;
+        this.uploadDocumentService = uploadDocumentService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .errors(validateGroupService.validateGroup(caseData, C21CaseOrderGroup.class))
+            .build();
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(
+        @RequestHeader(value = "authorization") String authorization,
+        @RequestHeader(value = "user-id") String userId,
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        Document c21Document = getDocument(authorization, userId, caseData);
+
+        caseDetails.getData().put("c21Order", service.addDocumentToC21(caseData.getC21Order(), c21Document));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<Element<C21Order>> c21Orders = caseData.getC21Orders();
+        c21Orders.add(Element.<C21Order>builder()
+            .id(randomUUID())
+            .value(service.addCustomValuesToC21Order(caseData.getC21Order(), caseData.getJudgeAndLegalAdvisor()))
+            .build());
```

Comment:
let's return and element from the service method and extract all of this... need to update tests etc